src/ima-middleware.js

```diff
@@ -97,6 +97,11 @@ class ImaMiddleware extends BaseMiddleware {
           this._callNextLoad();
         }
       });
+      this._context.player.addEventListener(this._context.player.Event.AD_LOADED, event => {
```

Comment:
- we already added a listener in line 83
- do we want to have this logic for every platform ? did it work on platforms except samsung ?

Reply:
fixed. want to do it for all. it reproduced on all platforms by disableMediaPreload option